### PR TITLE
Use Linebender CI

### PR DESCRIPTION
**Pull request by waywardmonkeys**
This also sets up the copyright headers on all files.

The MSRV is now set to 1.65. A minimum of 1.60 was needed for `log` and 1.62 for using `x86_64-unknown-none` for the `no_std` CI. `usvg` claims to have an MSRV of 1.65 (although a later version is actually tested).

Fix up the .gitignore while we're updating it.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,3 @@
 target
-Cargo.lock
 .idea
 *.iml
```
Comment (DJMcNab) [inline .gitignore:3]:
```suggestion
.DS_Store
```

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,3 @@
 target
```
Comment (DJMcNab) [inline .gitignore:1]:
```suggestion
/target
```

**Review by DJMcNab — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (waywardmonkeys) [reply .gitignore:3]: I don't think `.DS_Store` belongs in every single `.gitignore` and should be global.

Comment (DJMcNab) [reply .gitignore:3]: Sure. We're not likely to get any contributions to this repo, so I guess it's fine to assume they'll be competent.

**Pull request merged**